src/app/api/quality_matrix/models.py

```diff
@@ -1,8 +1,38 @@
+from typing import Any
+
 from pydantic import BaseModel, Field
+from sqlalchemy import JSON, Column, Integer, MetaData, Table
+from sqlalchemy.orm import as_declarative, declared_attr
 
 
 class ColumnOutputModel(BaseModel):
     metadatum: str = Field(default="", description="Name of the evaluated metadatum.")
     columns: dict[str, float] = Field(
         description="The ratio of quality for the required columns."
     )
+
+
+@as_declarative()
+class Base:
+    id: Any
+    __name__: str
+
+    @declared_attr
+    def __tablename__(cls) -> str:
+        return cls.__name__.lower()
+
+
+def timeline_table(meta: MetaData) -> Table:
```

Comment:
this violates DRY, according to [here](https://docs.sqlalchemy.org/en/14/orm/declarative_tables.html) you can simply get the table objet with `Timeline.__table__`. Also, I do not understand the `Base` class setup above why is it needed? Is it?